internal/protocoltest/awsrestjson/deserializers.go

```diff
@@ -686,69 +418,69 @@ func (m *awsRestjson1_deserializeOpJsonMaps) HandleDeserialize(ctx context.Conte
 	return out, metadata, err
 }
 
-func awsRestjson1_deserializeOpDocumentJsonMapsOutput(v *JsonMapsOutput, decoder *json.Decoder) (bool, error) {
+func awsRestjson1_deserializeOpDocumentJsonMapsOutput(v **JsonMapsOutput, decoder *json.Decoder) error {
 	if v == nil {
-		return false, fmt.Errorf("unsupported deserialization of nil %T", v)
+		return fmt.Errorf("unexpected nil of type %T", v)
 	}
 
 	startToken, err := decoder.Token()
 	if err == io.EOF {
-		return false, nil
+		return nil
 	}
 	if err != nil {
-		return false, err
+		return err
 	}
 	if startToken == nil {
-		return false, nil
+		return nil
 	}
 	if t, ok := startToken.(json.Delim); !ok || t.String() != "{" {
-		return false, fmt.Errorf("expect `{` as start token")
+		return fmt.Errorf("expect `{` as start token")
 	}
```

Comment:
Outside the scope of this PR, but just noticed this. All of these start/end token comparisons can do value comparison instead of converting the delim rune to a string.

```go
if t, ok := startToken, (json.Delim); !ok || t != json.Delim('{') {
```

Minor, but may save the additional allocations during deserialization turning the rune into a string potentially.